extensions/machine-learning-services/src/test/dialogs/externalLanguages/addEditLanguageTab.test.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as should from 'should';
+import 'mocha';
+import { createContext, ParentDialog } from './utils';
+import { AddEditLanguageTab } from '../../../dialogs/externalLanguages/addEditLanguageTab';
+import { LanguageUpdateModel } from '../../../dialogs/externalLanguages/languageViewBase';
+
+describe('Add Edit External Languages Tab', () => {
+	it('Should create AddEditLanguageTab for new language successfully ', async function (): Promise<void> {
+		let testContext = createContext();
+		let parent = new ParentDialog(testContext.apiWrapper.object);
+		let languageUpdateModel: LanguageUpdateModel = {
+			content: parent.createNewContent(),
+			language: parent.createNewLanguage(),
+			newLang: true
+		};
+		let tab = new AddEditLanguageTab(testContext.apiWrapper.object, parent, languageUpdateModel);
+		should.notEqual(tab.languageView, undefined);
+	});
+
+	it('Should create AddEditLanguageTab for edit successfully ', async function (): Promise<void> {
+		let testContext = createContext();
+		let parent = new ParentDialog(testContext.apiWrapper.object);
+		let languageUpdateModel: LanguageUpdateModel = {
+			content: {
+				extensionFileName: 'filename',
+				isLocalFile: true,
+				pathToExtension: 'path',
+			},
+			language: {
+				name: 'name',
+				contents: []
+			},
+			newLang: false
+		};
+		let tab = new AddEditLanguageTab(testContext.apiWrapper.object, parent, languageUpdateModel);
+		should.notEqual(tab.languageView, undefined);
```

Comment:
When you have multiple tests like this please make sure to add a message indicating what the failure is since it's not always clear. (This is helpful in general anyways so it's clear from the logs what the issue is)